docs/extensibility.md

```diff
@@ -0,0 +1,122 @@
+# Extensibility
+This guide provides an overview of extensibility options focusing on add-on extensions as the primary mechanism for the partners and customers. 
+## Overview
+SSP Framework is designed to be extensible. In the context of this guide, extensibility refers to the ability of customers and partners to both add new capabilities to the framework or platforms based on SSP as well as customize existing behavior, including ability to modify or override existing behavior.
+
+The following abstractions can be leveraged to add new features to the framework:
+
+- [Add-on](./core-concepts.md#add-on). Customers and partners can implement new add-ons which could be leveraged exactly the same way as the core add-ons (supplied by the framework).
+- [Resource Provider](./core-concepts.md#resource-provider). This construct allows to create resources that can be reused across multiple add-ons and/or teams. For example, IAM roles, VPC, hosted zone. 
```

Comment:
This construct allows customers to create

Reply:
addressed